Lib/idlelib/pyshell.py

```diff
@@ -1273,7 +1330,14 @@ def restart_shell(self, event=None):
 
     def showprompt(self):
         self.resetoutput()
-        self.console.write(self.prompt)
+
+        prompt = self.prompt
+        if self.sys_ps1 and prompt.endswith(self.sys_ps1):
+            prompt = prompt[:-len(self.sys_ps1)]
```

Comment:
This is pretty useless.  PyShell is initialized with
```
        self.sys_ps1 = sys.ps1 if hasattr(sys, 'ps1') else '>>> '
        self.prompt_last_line = self.sys_ps1.split('\n')[-1]
        self.prompt = self.sys_ps1  # Changes when debug active
```
The only way I know to either reset or delete sys.ps1 and start IDLE in the same process is to do so in python code and then `import idlelib.idle`.  Under normal circumstances, self.sys_ps1 exists, is not '', and and after the following code, equals self.prompt and self.prompt_last_line.

The new conditional will trigger once and clear the prompt to '' (or remove it from 'DEBUG ON\n'.   The conditional will never trigger again.  After it is cleared, the only possible 'prompt' within the text box is "DEBUG ON\n" .  (I don't think it should be a repeated prompt -- another issue.)  So we should remove sys_ps1.  Until we remove the debug prompt, we should just initialize self.prompt it to ''.

self.prompt_last_line is more problematical.  It should be set to '' or better deleted, but it is used (only) in EditorWindow to indicate "isinstance(self, PyShell)" and affect backspace and HOME.  (The assumption is that it is never '' in Shell, but it will be if sys.ps1 ends with '\n'.)  The uses of prompt_last_line need to be reviewed to remove uses of the string as a string and replace it with a boolean where the indicator is still needed.  (I started on this.)

Reply:
Indeed, this is needed to support custom prompts, as currently used only by the debugger (and possibly by extensions).

I was aiming to change as little as possible. The places where this is used are rather delicate, and changing them could requiring making other related changes... @terryjreedy, are you sure you want this as part of this PR?